src/components/Todos/index.js

```diff
@@ -7,17 +7,41 @@ import deserialize from './storage-source';
 import serialize from './storage-sink';
 import TodoItem from '../TodoItem';
 
+// AMEND STATE WITH CHILDREN
+// This is a callback function for a map function used below.
 function amendStateWithChildren(DOMSource) {
+  // For every todo item take `todosData`...
   return function (todosData) {
+    // ...and return stream data as an object
     return {
+      // Via the spread operator we copy all properties
+      // from todosData into the object we return minus
+      // the properties defined in the new object that have
+      // the same key as a property in todosData.
       ...todosData,
+      // The list property is the only property we are amending.
+      // For every item in todosData...
       list: todosData.list.map(data => {
+        // ...we create an Observable for the data item...
         const props$ = Observable.just(data);
+        // ...and create a new todo item by passing the DOM stream and
+        // the new props stream to it. The isolate function creates a
+        // "scoped data flow component" out of every todo item.
+        // For more read the docs here: https://github.com/cyclejs/isolate.
         const todoItem = isolate(TodoItem)({DOM: DOMSource, props$});
+        // Here we return the new amended array data for the list array.
         return {
+          // We copy the item data into the the returned object
+          // via the spread operator here...
           ...data,
+          // ...and add a new property called todoItem which
+          // returns a todoItem DOM stream sink and a todo item
+          // action stream sink.
           todoItem: {
             DOM: todoItem.DOM,
+            // The action stream for the todo item is enhanced with
+            // the id of the todo, in order to tell which item these actions
+            // are streaming from.
```

Comment:
I wouldn't merge any of these comments above in this function, because they are just paraphrasing the code. These types of comments don't really add value, they just re-explain the code, but add a lot of visual pollution. I would only add one line of comment to explain the **purpose** of this function: why it exists.